lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -570,6 +595,8 @@ def _run_script(self, rerun_data: RerunData) -> None:
         finally:
             if rerun_exception_data:
                 finished_event = ScriptRunnerEvent.SCRIPT_STOPPED_FOR_RERUN
```

Comment:
Can there also be the case in which a fragment run is stopped for rerun and, if yes, should we have a dedicated finished event for it similar to the distinct finish event `FRAGMENT_STOPPED_WITH_SUCCESS`?

Reply:
Fragments can be stopped early for a rerun, but the handling should be identical to when a normal script run is stopped early (the `AppSession` class handles this message type by changing its state to `APP_NOT_RUNNING` and sending a message to the client to acknowledge that it's stopping). Since the handling is identical, we probably don't want a new event for this, but I'll add an explanatory comment here to point this out